File: 0424-longest-repeating-character-replacement/0424-longest-repeating-character-replacement.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        int n = s.size();
        int r = 0, l = 0, max_f = 0, max_len = 0;
        vector<int> v(26,0);
        while(r < n) {
            v[s[r] - 'A']++;
            max_f = max(max_f,v[s[r] - 'A']); 
            while((r - l + 1) - max_f > k) {
                v[s[l] - 'A']--;
                max_f = 0;
                for(int i=0; i<26; i++) {
                    max_f = max(max_f,v[i]);
                }
                l++;
            }
            if((r - l + 1) - max_f <= k) max_len = max(max_len,(r-l+1));
            r++;
        }
        return max_len;
    }
};

File: 0424-longest-repeating-character-replacement/0424-longest-repeating-character-replacement.java
class Solution {
    public int characterReplacement(String s, int k) {
        int count = 0;
        int max_count = 0;
        int max_len = 0;
        int i = 0;
        int[] arr = new int[26];
        for(int j = 0; j < s.length(); j++) {
            arr[s.charAt(j) - 'A']++;
            max_count = Math.max(max_count, arr[s.charAt(j) - 'A']);
            count = j - i + 1;
            while(count - max_count > k) {
                arr[s.charAt(i) - 'A']--;
                i++;
                count = j - i + 1;
            }
            max_len = Math.max(max_len, count);
        }
        return max_len;
    }
}